site/app/models/GradeableTestcase.php

```diff
@@ -150,6 +164,7 @@ public function getAutochecks() {
     }
     
     public function hasDetails() {
+        if ($this->isHidden()) return false;
         return count($this->autochecks) > 0 || $this->hasCompilationOutput() || $this->hasExecuteLog();
```

Comment:
I think the long line of checks with ands, ors, and nots is harder to read and maintain.
Anyhow, I've cleaned up some dead code so it's your kind of one-liner now.